### PR TITLE
add support for .pptx, .xlsx format

### LINKED ISSUE #12: support (.pptx) format
#### Description
Capability to remove Microsoft PowerPoint files metadata

### LINKED ISSUE #13: support (.xlsx) format
#### Description
Capability to remove Microsoft Excel files metadata

### PR DESCRIPTION
**Pull request by AHReccese**
#### Reference Issues/PRs
#12 #13 
#### What does this implement/fix? Explain your changes.
- replace XML parser with `lxml` parser of the `defusedxml` library
- add support for both .pptx and .xlsx formats
- refactorings and generalizing functions

### CODE REVIEW

File: dmeta/functions.py [hunk 1]
```diff
@@ -4,75 +4,75 @@
...
     dir_list = os.listdir(path)
-    docx_files = []
+    microsoft_files = []
     for item in dir_list:
-        if ".docx" in item:
-            docx_files.append(item)
-    for docx_file in docx_files:
-        clear(docx_file)
+        if get_microsoft_format(item) is not None:
+            microsoft_files.append(item)
```
Comment (sadrasabouri) [inline dmeta/functions.py:63]: and handle the exception here.

**Review by sadrasabouri — CHANGES_REQUESTED** [review]
It looks good to me.
I tested it and it worked correctly. I left a minor comment.

**Comment by AHReccese** [comment]
Dear Sadra, I will handle errors and exceptions in the next PR. If it's possible, skip it in this PR 
@sadrasabouri

File: dmeta/util.py [hunk 2]
```diff
@@ -18,43 +19,43 @@ def extract_namespaces(xml_file_path):
...
-    last_dot_index = docx_file_name.rfind('.')
-    if (last_dot_index != -1):
-        docx_file_name = docx_file_name[:last_dot_index]
-    return docx_file_name
+    last_dot_index = file_name.rfind('.')
+    if (last_dot_index == -1):
+        return None
+    format = file_name[last_dot_index + 1:]
+    if format not in SUPPORTED_MICROSOFT_FORMATS:
+        return None
```
[the remark this answers is not in the document]
Comment (AHReccese) [reply dmeta/util.py:45]: I will handle all exceptions and related craps in the serious in detail PR soon.
I will define custom DMeta-defined errors same as our other projects.

**Pull request merged**